src/components/gridListItem/index.tsx

```diff
@@ -139,14 +152,35 @@ class GridListItem extends Component<GridListItemProps> {
     return {width: itemSize as number, height: itemSize as number};
   }
 
+  getHorizontalAlignmentStyle = memoize(horizontalAlignment => {
+    switch (horizontalAlignment) {
+      case HorizontalAlignment.left:
+        return {contentStyle: styles.contentAlignedToStart, containerStyle: styles.containerAlignedToStart};
+      case HorizontalAlignment.right:
+        return {contentStyle: styles.contentAlignedToEnd, containerStyle: styles.containerAlignedToEnd};
+      case HorizontalAlignment.center:
+        return {contentStyle: styles.contentAlignedToCenter, containerStyle: styles.containerAlignedToCenter};
+      default:
+        undefined;
+    }
+  });
+
   renderContent({text, typography, color, numberOfLines = 1, style, testID}: RenderContentType) {
-    const {alignToStart} = this.props;
+    const {alignToStart, horizontalAlignment} = this.props;
+    const horizontalAlignmentStyle = this.getHorizontalAlignmentStyle(horizontalAlignment);
     if (text) {
       return (
         <Text
           testID={testID}
           // @ts-ignore
-          style={[style, Typography[typography], color && {color}, alignToStart && styles.contentAlignedToStart]}
+          style={[
+            style,
+            //@ts-ignore
+            Typography[typography],
+            color && {color},
+            alignToStart && styles.contentAlignedToStart,
+            horizontalAlignmentStyle?.contentStyle
```

Comment:
{textAlign: HorizontalAlignment} - can replace this object, just place it before this line: 
`alignToStart && styles.contentAlignedToStart,` to avoid overriding it